File: src/main/java/cn/xunyard/idea/coding/log/LoggerFactory.java
package cn.xunyard.idea.coding.log;

import cn.xunyard.idea.coding.log.window.WindowLogger;

import java.util.HashMap;
import java.util.Map;

/**
 * @author <a herf="mailto:wuqi@terminus.io">xunyard</a>
 * @date 2019-12-26
 */
public class LoggerFactory {

    private static Map<String, Logger> loggerMap;

    public static Logger getLogger(String identityId) {
        if (loggerMap == null) {
            loggerMap = new HashMap<>();
        }

        if (loggerMap.containsKey(identityId)) {
            return loggerMap.get(identityId);
        }

        WindowLogger logger = new WindowLogger(identityId);
        loggerMap.put(identityId, logger);
        return logger;
    }

}

File: src/main/kotlin/cn/xunyard/idea/coding/log/LoggerFactory.kt
package cn.xunyard.idea.coding.log

import cn.xunyard.idea.coding.log.window.WindowLogger

/**
 * @author <a herf="mailto:xunyard@gmail.com">xunyard</a>
 * @date 2019-12-26
 */
object LoggerFactory {
    private val loggerMap: MutableMap<String, Logger> = HashMap()

    fun getLogger(identityId: String): Logger {
        if (loggerMap.containsKey(identityId)) {
            return loggerMap[identityId] ?: throw IllegalArgumentException("invalid identity of $identityId")
        }
        val logger = WindowLogger(identityId)
        loggerMap[identityId] = logger
        return logger
    }
}